modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/WorkspaceExtension.java

```diff
@@ -341,6 +357,10 @@ public Plugin<Project> getRootProjectConfigurator() {
 	}
 
 	public String getTargetPlatformVersion() {
+		if (Objects.isNull(_targetPlatformVersion)) {
+			return _getDefaultTargetplatformVersion();
+		}
```

Comment:
@simonjhy what causes these fields to be null? Aren't they set in the constructor with fallbacks to the default methods (like `_getDefaultTargetplatformVersion()`)?

Reply:
1) Firstly customer can remove all itemes defined in gradle.properties and we also support customer to only define them in build.gradle. Like this  
    gradle.liferayWorkspace {
	    product = "portal-7.4-ga8"
    }
    when this happened, although we set _targetPlatformVersion in the constructor of workspaceExtension, but the workspaceExtension will be initialized before gradle evaluated event start, gradle will not konw what's value of product key user defined by in build.gradle.  At this time the product key is null and the _targetPlatformVersion is null too. So the _targetPlatformVersion will be set null and _getDefaultTargetplatformVersion also reture null. 

2) When gradle evaluated event finished, gradle will get the prodcut value custoemr define in build.gradle. But the old 
getTargetPlatformVersion method just return the _targetPlatformVersion variable, this variable will be null for ever because we did not directly define targetplatformVersion in build.gradle, the setTargetPlatformVersion method will be never called. Actually, we should get correct value from productInfo.json because customer set product key at this moment.